### ISSUE TITLE
Deploy to Vercel - Vercel Builder or Adapter

### ISSUE THREAD

**Issue by jdgamble555** [angular/universal#2486]
# 🚀 Feature request

#### What modules are relevant for this feature request?

- [ ] aspnetcore-engine
- [X ] builders
- [X ] common
- [ X] express-engine
- [ ] hapi-engine

### Description

I would like the ability to deploy an Angular Universal app to Vercel.  

### Describe the solution you'd like

It seems vercel has an adapter for NuxtJs, NextJs, Sapper, and Sveltekit.  At the very least, I would think something like this in a vercel.json file would work:

```typescript
{
  "version": 2,
  "public": true,
  "name": "test-universal",
  "builds": [
    { "src": "package.json", "use": "@vercel/static-build" },
    { "src": "./dist/test/server/main.js", "use": "@vercel/node" }
  ],
  "routes": [{ "src": "/(.*)", "dest": "./dist/test/server/main.js" }]
}
```
**SPOILER** - This does not currently work.

### Describe alternatives you've considered

I am not sure whether or not it should work with or without a custom adapter.  If we can put an Angular Universal app in Cloud Run or Firebase Functions, it should work on a Vercel Lambda Function the same way all other frameworks do.   It should be so easy to do.

Vercel used to be Zeit.  So here are some relevant links:

- https://github.com/enten/angular-universal/issues/26
- https://medium.com/@fransyozef/deploy-a-nestjs-application-with-hosted-prisma-database-on-zeit-now-for-free-631f0de87486
- https://github.com/nestjs/nest/issues/8853#issuecomment-1000953137

J

**Comment by damienwebdev** [comment]
This is possible -- here's how I do it:

https://github.com/graycoreio/github-actions/tree/main/angular-universal-vercel

Example websites:

https://next.graycore.io
https://next.daff.io
https://next.mappia.io

**Comment by jdgamble555** [comment]
So I actually wrote an article on how to do what you did [How to Deploy Angular Universal to Vercel](https://dev.to/jdgamble555/how-to-deploy-angular-universal-to-vercel-31d0).  Your code is better :)

I think there needs to be a way to put this inside of `ng deploy` as well as an actual [Vercel Integration](https://vercel.com/integrations).  Vercel has told me when I contacted them a while ago, that they don't build adapters, but they will put the adapters on the official Vercel site when they are built.  

https://github.com/sveltejs/kit/tree/master/packages/adapter-vercel
https://github.com/nuxt/vercel-builder

Of course the SvelteKit Adapter was built before Vercel acquired Svelte.

J
